eZ/Publish/API/Repository/Tests/SettingServiceTest.php

```diff
@@ -0,0 +1,217 @@
+        /** @var \eZ\Publish\API\Repository\SettingService $settingService */
+        $settingService = $container->get('ezpublish.api.service.setting');
+
+        return $settingService;
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $repository = $this->getRepository(false);
+        $this->permissionResolver = $repository->getPermissionResolver();
+    }
+
+    /**
+     * Test for the newSettingCreateStruct() method.
+     *
+     * @see \eZ\Publish\API\Repository\SettingService::newSettingCreateStruct()
+     */
+    public function testNewSettingCreateStruct()
+    {
+        $settingService = $this->getSettingService();
+        $settingCreate = $settingService->newSettingCreateStruct();
+
+        $this->assertInstanceOf(SettingCreateStruct::class, $settingCreate);
+    }
+
+    /**
+     * @see \eZ\Publish\API\Repository\SettingService::createSetting()
+     */
+    public function testCreateSetting()
+    {
+        $settingService = $this->getSettingService();
+
+        $settingCreate = $settingService->newSettingCreateStruct();
+        $settingCreate->setGroup('test_group');
+        $settingCreate->setIdentifier('test_identifier');
+        $settingCreate->setValue('test_value');
+
+        $setting = $settingService->createSetting($settingCreate);
+
+        $this->assertInstanceOf(Setting::class, $setting);
+    }
+
+    /**
+     * @see \eZ\Publish\API\Repository\SettingService::createSetting()
+     */
+    public function testCreateSettingThrowsInvalidArgumentException()
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        $settingService = $this->getSettingService();
+
+        $settingCreateFirst = $settingService->newSettingCreateStruct();
+        $settingCreateFirst->setGroup('test_group');
+        $settingCreateFirst->setIdentifier('test_identifier');
+        $settingCreateFirst->setValue('test_value');
+
+        $settingService->createSetting($settingCreateFirst);
+
+        $settingCreateSecond = $settingService->newSettingCreateStruct();
+        $settingCreateSecond->setGroup('test_group');
+        $settingCreateSecond->setIdentifier('test_identifier');
+        $settingCreateSecond->setValue('another_value');
+
+        $settingService->createSetting($settingCreateSecond);
+    }
+
+    /**
+     * @see \eZ\Publish\API\Repository\SettingService::loadSetting()
+     */
+    public function testLoadSetting()
+    {
+        $settingService = $this->getSettingService();
+
+        $settingCreate = $settingService->newSettingCreateStruct();
+        $settingCreate->setGroup('another_group');
+        $settingCreate->setIdentifier('another_identifier');
+        $settingCreate->setValue('test_value');
+
+        $settingService->createSetting($settingCreate);
+
+        $setting = $settingService->loadSetting('another_group', 'another_identifier');
+
+        $this->assertInstanceOf(Setting::class, $setting);
```

Comment:
Provides no value, it's always gonna be either `Setting` or fatal error prior that.